pkg/liqoctl/install/install_test.go

```diff
@@ -0,0 +1,254 @@
+import (
+	"reflect"
+	"testing"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/ginkgo/extensions/table"
+	. "github.com/onsi/gomega"
+	"github.com/spf13/cobra"
+	"helm.sh/helm/v3/pkg/action"
+
+	liqoconst "github.com/liqotech/liqo/pkg/consts"
+	installprovider "github.com/liqotech/liqo/pkg/liqoctl/install/provider"
+	installutils "github.com/liqotech/liqo/pkg/liqoctl/install/utils"
+)
+
+type StringInterfaceMap map[string]interface{}
+
+// IsContainedIn checks whether specMap's key-value pairs are equal to testMap's ones (not the other way around).
+func (specMap StringInterfaceMap) IsContainedIn(testMap map[string]interface{}) bool {
+	isContained := false
+
+	for key := range specMap {
+		testValue, ok := testMap[key]
+		if !ok {
+			break
+		}
+		specValue, ok := specMap[key]
+		if !ok {
+			break
+		}
+
+		switch {
+		case reflect.TypeOf(testValue) != reflect.TypeOf(specValue):
+			// Do nothing
+		case reflect.TypeOf(testValue).Kind() == reflect.Map:
+			specValue := StringInterfaceMap(specValue.(map[string]interface{}))
+			testValue := testValue.(map[string]interface{})
+			isContained = specValue.IsContainedIn(testValue)
+		case reflect.TypeOf(testValue).Kind() == reflect.Slice:
+			specValue := specValue.([]interface{})
+			testValue := testValue.([]interface{})
+			if len(specValue) == len(testValue) {
+				isContained = true
+				for i := range specValue {
+					isContained = testValue[i] == specValue[i]
+					if !isContained {
+						break
+					}
+				}
+			}
+		default:
+			isContained = testValue == specValue
+		}
+
+		if !isContained {
+			break
+		}
+	}
+
+	return isContained
+}
+
+func TestInstallCommand(t *testing.T) {
+	RegisterFailHandler(Fail)
+	t.TempDir()
+	RunSpecs(t, "Test Install Command")
+}
+
+var _ = Describe("Test the install command works as expected", func() {
+
+	type testCase struct {
+		provider       string
+		parameters     []string
+		expectedValues StringInterfaceMap
+	}
+
+	DescribeTable("An install command is issued",
+		func(tc testCase) {
+			cmd := &cobra.Command{}
+			cmd.PersistentFlags().Bool("debug", false, "")
+			cmd.PersistentFlags().Int("timeout", 600, "")
+			cmd.PersistentFlags().String("version", "", "")
+			cmd.PersistentFlags().Bool("devel", false, "")
+			cmd.PersistentFlags().Bool("only-output-values", false, "")
+			cmd.PersistentFlags().String("dump-values-path", "", "")
+			cmd.PersistentFlags().Bool("dry-run", false, "")
+			cmd.PersistentFlags().Bool("enable-lan-discovery", false, "")
+			cmd.PersistentFlags().String("cluster-labels", "", "")
+			cmd.PersistentFlags().Bool("disable-endpoint-check", false, "")
+			cmd.PersistentFlags().String("chart-path", installutils.LiqoChartFullName, "")
+			cmd.PersistentFlags().String("cluster-name", "", "")
+			cmd.PersistentFlags().String("reserved-subnets", "", "")
+			cmd.PersistentFlags().String("resource-sharing-percentage", "90", "")
+			cmd.PersistentFlags().Bool("enable-ha", false, "")
+			cmd.PersistentFlags().Int("mtu", 0, "")
```

Comment:
Wondering why it is necessary to redefine all arguments here.

Reply:
I removed the unnecessary calls to external functions and therefore the definition of many of those arguments was no more required.